packages/code-studio/vite.config.ts

```diff
@@ -26,13 +26,14 @@ export default defineConfig(({ mode }) => {
     port = 4000;
   }
 
+  const baseURL = new URL(env.BASE_URL, `http://localhost:${port}/`);
   // These are paths which should be proxied to the core server
   // https://vitejs.dev/config/server-options.html#server-proxy
   const proxy = {
     // Proxy styleguide here instead of as a route in our app router
     // That way, it is not included in the production build
-    '/styleguide': {
-      target: `http://localhost:${port}/src/styleguide/index.html`,
+    '^/styleguide': {
```

Comment:
You don't need the caret here really. Not a regex and matches start of the path. Shouldn't hurt anything though. [Docs](https://vitejs.dev/config/server-options.html#server-proxy)